test/e2e/operator/nmstate_install_test.go

```diff
@@ -0,0 +1,110 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	nmstatev1beta1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1beta1"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	v1 "k8s.io/api/apps/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	nmstate     nmstatev1beta1.NMState
+	nmstateKey  = types.NamespacedName{Namespace: "nmstate", Name: "nmstate"}
+	handlerKey  = types.NamespacedName{Namespace: "nmstate", Name: "nmstate-handler"}
+	operatorKey = types.NamespacedName{Namespace: "nmstate", Name: "nmstate-operator"}
+	webhookKey  = types.NamespacedName{Namespace: "nmstate", Name: "nmstate-webhook"}
```

Comment:
This info is only partial. You keep the key here, but you also have to know it is a key of a webhook. Maybe you could define helper functions for these `gets` that would carry both the knowledge of the name and the type?

Reply:
I'm going to add one, and see if I'm understanding what you had in mind. If you like it, I'll do the others.

Reply:
I mean a function ~`getNMStateHandler` that knows the name of the handler and its type, so it just calls the API to obtain it and returns it. Now totally sure it would fit into the rest of testing.

Reply:
I think I'm grokking what you're saying. Take a look at the latest patch?